macros.go

```diff
@@ -127,6 +133,49 @@ func trimAll(s []string) []string {
 	return r
 }
 
+var pair = map[rune]rune{')': '('}
+
+// parseInput find the closing parenthesis of the macro string and adds an empty space,
+// so that our macro regex test has an unambiguous result (because regex has no memory)
+func parseInput(input string, name string) string {
```

Comment:
since you already have a function that finds the closing parenthesis of the macro, I think we can just skip the regex (and also the spaces workaround). If this function returns an array with the whole macro and the arguments (if exists), e.g. `["$__foo(bar)", "bar"]` you can just return that as the result for `getMatches`

Reply:
@andresmgot  Well the problem is that this only finds the macro,but not the arguments. Since I already have the macro string I was trying to return this in a form of `$__macro(<regex_for_capturing_group>)` so that regex can take care of the arguments part, but in no way go would accept that. And since I don't know all the possible cases I was weary to just treat everything between the parentheses as argument.

But if it's guaranteed, that `$__macro(<argument>)` is always treated as a single argument, I can just modify this function to work as `regex.FindAllStringSubmatch` and return an array with match and argument.

Reply:
yes, this function treats it as a single argument (that's what the regex does). Then, it's split by `,` in the `Interpolate` function:

```go
			if len(match) > 1 {
				// This macro has arguments
				args = trimAll(strings.Split(match[1], ","))
			}
```

So it's safe to do that assumption

Reply:
I believe that's because it's what the regex returned (but it's fine to leave this as it is)